### ISSUE TITLE
Rule proposal: `prefer-static-response-json` instead of `new Response(JSON.stringify())`

### ISSUE THREAD

**Issue by regseb** [sindresorhus/eslint-plugin-unicorn#1960]
### Description

Prefer the static [`Response.json()`](https://fetch.spec.whatwg.org/#dom-response-json) method instead of `new Response(JSON.stringify())`.

More info: https://github.com/whatwg/fetch/issues/1389 / https://github.com/whatwg/fetch/pull/1392

### Fail

```js
const response = new Response(JSON.stringify({ foo: "bar" }));
```

### Pass

```js
const response = Response.json({ foo: "bar" });
```

**Comment by sindresorhus** [comment]
The method is quite new, so the rule cannot be in the recommended preset, and we should add a note in the docs about it being quite new.

**Comment by fisker** [comment]
This pattern seems rarely used.

https://github.com/search?q=new+Response%28JSON.stringify&type=code&p=1

**Comment by silverwind** [comment]
For what purpose are you `Response` like that? I don't see the use case.

**Comment by regseb** [comment]
There are two `json()` methods in the `Response` class:

- `static Response.json(data, init)`: Create a response that contains JSON.
- `Response.prototype.json()`: Parse the JSON of the response.

---

In my unit tests, I mock the fetch method to return a specific JSON (examples in [Cast Kodi](https://github.com/regseb/castkodi/search?q=new+Response%28JSON.stringify%28)):

```javascript
it("should return video URL", async function () {
    const stub = sinon.stub(globalThis, "fetch").resolves(new Response(
        JSON.stringify({
            media: [{
                id:          "foo",
                offline_url: "http://bar.com/baz.mp4",
            }],
        }),
    ));

    const file = await scraper.extract("https://watch.pokemon.com/player?id=foo");
    assert.equal(file, "http://bar.com/baz.mp4");

    assert.equal(stub.callCount, 1);
    assert.deepEqual(stub.firstCall.args, [
        "https://www.pokemon.com/api/pokemontv/v2/channels/uk",
    ]);
});
```

With this new rule, the above code would be in error. You would have to write:


```javascript
it("should return video URL", async function () {
    const stub = sinon.stub(globalThis, "fetch").resolves(Response.json({
        media: [{
            id:          "foo",
            offline_url: "http://bar.com/baz.mp4",
        }],
    }));

    // ...
});
```

**Comment by silverwind** [comment]
I see. Yes I suspected the primary use case for this is mocking. So I'm neutral on this one, don't see it much in demand, thought.

**Comment by regseb** [comment]
Deno says: "This addition to the Fetch specification was specifically designed with server side runtimes in mind." [_Deno 1.22 Release Notes - New `Response.json()` static method_](https://deno.com/blog/v1.22#new-responsejson-static-method).

For example:

```typescript
import { serve } from "https://deno.land/std@0.165.0/http/server.ts";

const port = 8080;

const handler = (request) => {
    const body = { userAgent: request.headers.get("user-agent") ?? "Unknown" };

    // Previously:
    // return new Response(JSON.stringify(body), {
    //     headers: { "content-type": "application/json" },
    // });

    // Now:
    return Response.json(body);
};

await serve(handler, { port });
```

**Comment by silverwind** [comment]
Yeah, for servers that support `Response` for actually responding, it's definitely useful.

What I'm [not fond of](https://github.com/whatwg/fetch/issues/1389#issuecomment-1346954967) is that the static method is named the same as the prototype method that is used to parse JSON on the client, but I guess that is not for this topic here.

**Comment by fisker** [comment]
Should we rename it as `prefer-response-static-json`?
If one day we want to add a rule for the instance `json` method, we definitely don't want use `prefer-{instance,prototype}-response-json`.
